### DISCUSSION TITLE
Data importer: Remove polling time limit (601 seconds)

### DISCUSSION THREAD

**Discussion by reredan** [firefly-iii/firefly-iii#9157] [Firefly III]
Hi there! I am trying to import a large dataset and the process seems to work properly. However, as the process takes a long time I am getting the following error "The data importer has been polling for more than 601 seconds. It has stopped, to prevent eternal loops". Is there any way of removing or extending that time limit? Thank you!

**Comment by oddstr13** [comment]
I'm hitting the same issue when importing a CSV file with a years worth of transactions.
The status of submitted transactions was updating once every second or so until the error message appeared, so it clearly wasn't stuck or infinite looping.

It almost got thru February, so it was almost done too (reverse sort in the file)! :disappointed:

**Comment by JC5** [comment]
Hi, thanks for the suggestion!

If you need to import large batches, please use the command line interface instead.

https://docs.firefly-iii.org/how-to/data-importer/advanced/cli/

**Reply by reredan** [reply]
@JC5 Thanks, this is very helpful - overlooked this option. It's great because it allows for directory import as well.

@oddstr13: I noticed that despite the error message I got, the process continued in the background and all of my data was uploaded correctly.

**Reply by JC5** [reply]
I want to give you a small heads-up that this is not always the case, the PHP process may get killed unexpectedly. This depends on your setup, although the Docker container usually continues unabated.

**Reply by oddstr13** [reply]
My import may have continued after the error, but it didn't complete.
I figured out the CLI process, which is faster, altho I'm still fighting with the importer on how to actually import the data provided by my bank properly.
(I've got from and to account fields, not this and other account, which seems to trip up the importer – I'm already massaging the data a little bit to fix encoding and split it up by year/month, I might just make it move the data around some too)

**Reply by JC5** [reply]
From and to accounts should not be a problem. Just set them to "asset" and "opposing" and the data importer will reverse them automatically, usually based on the amount.
